components/query_wait.py

```diff
@@ -14,6 +14,8 @@
 
 import time
 
+from multiprocessing import Event
```

Comment:
This is used instead of `time.sleep` due to some of the multiprocessing constraints found within the app time.sleep can introduce a hang. To avoid any potential issues we're now using an event delay.